client/go/parser_assetlist.go

```diff
@@ -0,0 +1,101 @@
+package assetlist
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+)
+
+var osmosisAssetList Root
+
+func init() {
+	// Read the content of the JSON file using ioutil.ReadFile().
+	relativePathAssetlist := filepath.Join("..", "..", "osmosis-1", "osmosis-1.assetlist.json")
+	file, err := ioutil.ReadFile(relativePathAssetlist)
+	if err != nil {
+		panic(err)
+	}
+
+	// Decode the JSON content using json.Unmarshal() and store it in a variable of type AssetList.
+	err = json.Unmarshal(file, &osmosisAssetList)
+	if err != nil {
+		panic(err)
+	}
+}
```

Comment:
This shouldn't be an init file. All of the functions should take in an assetlist struct (OR be methods on an assetlist object)

Reply:
Its up to the library user to parse the file pre-marshalling. (We could make a convenience method for this if we want)

Reply:
Yep, agree, I have add `ReadAssetListUnmarshal` function and unittest for `osmosis-1` and `osmo-4` assetlist, since I feel like it's more like duplicate code from decode_test.go, I have remove it.
We parse the file for user and return assetlist struct, each function use receiver pointer object to parse assetlist.
Let me know, if this is fine